core/new-gui/src/app/workspace/component/navigation/navigation.component.html

```diff
@@ -9,22 +9,49 @@
 
     <!-- workflow metadata display -->
     <div *ngIf="userSystemEnabled" class="texera-navigation-workflow-metadata">
+      <button
+        (click)="closeParticularVersionDisplay()"
+        *ngIf="displayParticularWorkflowVersion"
+        nz-button
+        nzType="text"
+        title="back"
+        class="particular-version-metadata"
+      >
+        <i nz-icon nzTheme="outline" nzType="arrow-left"></i>
+      </button>
       <label>
         <input
+          *ngIf="!displayParticularWorkflowVersion"
           (change)="onWorkflowNameChange()"
           [(ngModel)]="currentWorkflowName"
           class="workflow-name"
           placeholder="Untitled Workflow"
         />
+        <span *ngIf="displayParticularWorkflowVersion" class="particular-version-metadata" spellcheck="false">
+          {{particularVersionDate}}
+        </span>
       </label>
-      <span class="save-state" spellcheck="false" (click)="onClickGetAllVersions()">{{autoSaveState}}</span>
+      <button *ngIf="displayParticularWorkflowVersion" class="btn btn-primary btn-sm" (click)="revertToVersion()">
+        Restore this version
+      </button>
+      <button
+        *ngIf="!displayParticularWorkflowVersion"
+        nz-button
+        nzType="link"
+        title="Open version history"
+        class="save-state"
+        spellcheck="false"
+        (click)="onClickGetAllVersions()"
+      >
+        {{autoSaveState}}
+      </button>
```

Comment:
as the logic becomes larger (switching between multiple components) and overlaps some other components, can we extract this part out to be a separate component?

Reply:
good idea. maybe we can do that in a future refactoring PR to also remove the 'user system' related buttons also from the navigation component

Reply:
yeah the navigation component urgently needs a refactoring. I feel if you can do the extraction of `version` related logic now it would be good. otherwise we are just adding more mixed code into this pot and make it harder to extract later.

Reply:
If this is the only comment, I suggest we merge this PR because it is blocking other tasks. Refactoring `navigation` component shall be done carefully and at once to include the more pressing issue of extracting `user system` logic not only the `version control` such that the effort is done once and not wasted.